### PR TITLE
Add examples for HTTP-to-HTTPS redirects.

### LINKED ISSUE #2622: Docs: Add full example of HTTP->HTTPS redirect
**What would you like to be added**:
Addition to https://gateway-api.sigs.k8s.io/guides/http-redirect-rewrite/#redirects that includes the following:

1. A Gateway with HTTP and HTTPS listeners
2. A HTTPRoute attached to the HTTP listener that does the redirect to HTTPS
3. A simple HTTPRoute attached to the HTTPS listener that forwards to any sample application

**Why this is needed**:
Currently our redirect docs have a [very basic redirect example](https://gateway-api.sigs.k8s.io/guides/http-redirect-rewrite/#redirects), but they leave out some very important details. This was made especially clear in https://github.com/kubernetes-sigs/gateway-api/issues/1185#issuecomment-1832955997 (thanks @HummingMind).

### PR DESCRIPTION
**Pull request by sawsa307**
* Add example Gateway and HTTPRoute yamls for configuring HTTP-to-HTTPS redirect for Gateway.
* Rename httproute-redirect-https.yaml to httproute-redirect-http.yaml since the redirect HTTPRoute only applies on HTTP listener. Update all reference to this file.
* Include a HTTPRoute for HTTPS traffic handling in httproute-redirect-https.yaml.

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add examples for HTTP-to-HTTPS redirects.

**Which issue(s) this PR fixes**:
Fixes #2622 

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

File: examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - name: redirect-gateway
+    sectionName: http
+  hostnames:
+  - redirect.example
+  rules:
+    - filters:
+        - type: RequestRedirect
+          requestRedirect:
+            scheme: https
+            statusCode: 301
```
Comment (robscott) [inline examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml:18]: Nit: Recommend updating indentation to match what you've done elsewhere.
```suggestion
  - filters:
    - type: RequestRedirect
      requestRedirect:
        scheme: https
        statusCode: 301
```

File: examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#$ Used in:
+#$ - site-src/api-types/httproute.md
```
Comment (robscott) [inline examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml:2]:
```suggestion
#$ - site-src/guides/http-redirect-rewrite.md
```

File: examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml [hunk 3]
```diff
@@ -1,15 +1,18 @@
 #$ Used in:
 #$ - site-src/api-types/httproute.md
```
Comment (robscott) [inline examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml:2]:
```suggestion
#$ - site-src/guides/http-redirect-rewrite.md
```

File: site-src/guides/http-redirect-rewrite.md [hunk 4]
```diff
@@ -31,6 +31,32 @@ example, the request `GET http://redirect.example/cinnamon` will result in a
...
+### HTTP-to-HTTPS redirects
+
+To redirect HTTP traffic to HTTPS, you need to have a Gateway with both HTTP
+and HTTPS listeners.
+
+```yaml
+{% include 'standard/http-redirect-rewrite/gateway-redirect-http-https.yaml' %}
+```
+There are multiple ways to secure a Gateway. In this example, it is secured
+using a Kubernetes Secret(`redirect-example` in `certificateRefs` section).
```
Comment (robscott) [inline site-src/guides/http-redirect-rewrite.md:43]:
```suggestion
using a Kubernetes Secret(`redirect-example` in the `certificateRefs` section).
```

File: examples/standard/http-redirect-rewrite/gateway-redirect-http-https.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#$ Used in:
+#$ - site-src/guides/http-redirect-rewrite.md
+kind: Gateway
+apiVersion: gateway.networking.k8s.io/v1
```
Comment (robscott) [inline examples/standard/http-redirect-rewrite/gateway-redirect-http-https.yaml:4]: Tiny nit, most k8s examples put `apiVersion` before `kind`.
```suggestion
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
```

File: examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml [hunk 6]
```diff
@@ -1,15 +1,18 @@
...
-apiVersion: gateway.networking.k8s.io/v1beta1
 kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1
```
Comment (robscott) [inline examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml:4]: Same comment as above
```suggestion
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
```

**Review by robscott — COMMENTED** [review]
Thanks @sawsa307! This will be a great improvement!

/ok-to-test

File: examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml (hunk 1 above)
Comment (sawsa307) [reply examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml]: Updated!

File: examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml (hunk 2 above)
Comment (sawsa307) [reply examples/standard/http-redirect-rewrite/httproute-redirect-http.yaml]: Updated!

File: examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml (hunk 3 above)
Comment (sawsa307) [reply examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml]: Updated!

File: examples/standard/http-redirect-rewrite/gateway-redirect-http-https.yaml (hunk 5 above)
Comment (sawsa307) [reply examples/standard/http-redirect-rewrite/gateway-redirect-http-https.yaml]: Updated!

File: examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml (hunk 6 above)
Comment (sawsa307) [reply examples/standard/http-redirect-rewrite/httproute-redirect-https.yaml]: Updated!

**Comment by sawsa307** [comment]
> Thanks @sawsa307! This will be a great improvement!
> 
> /ok-to-test

Addressed comments. Please take a look!

**Comment by robscott** [comment]
Thanks @sawsa307! Will leave final LGTM for someone else in case I missed something.

/cc @arkodg @candita @sunjayBhatia 
/approve

**Review by arkodg — APPROVED** [review]
/lgtm thanks !

**Comment by robscott** [comment]
Not sure why the first LGTM didn't work, adding another.

/lgtm

**Pull request merged**